jax/_src/numpy/ufunc_api.py

```diff
@@ -32,6 +33,35 @@
 import numpy as np
 
 
+def get_single_primitive(fun, *args):
```

Comment:
To ensure that the jaxpr represents exactly the application of the primitive, and isn't e.g. dropping or permuting inputs and outputs, maybe we should also check things like `jaxpr.invars == jaxpr.eqns[0].invars` and `jaxpr.eqns[0].outvars == jaxpr.outvars`.

Reply:
(Or maybe if all the fastpath ops are commutative, permutations are okay?)

Reply:
Good call - done!